### DISCUSSION TITLE
Decouple plugins entirely?

### DISCUSSION THREAD

**Discussion by dimus** [helix-editor/helix#9290] [Ideas] [answered]
In this suggestion, I wonder about Helix's development  path in the context of my experiences with Vim, NeoVim, and  a synergy I recently found in VS Code with NeoVim integration. I propose a strategic approach for Helix's future, considering the balance between maintaining its simplicity and providing advanced functionalities through potential auxiliary systems.

As a seasoned Vim user (20+ years), I've observed its diminishing relevance. NeoVim addresses the relevance issue by incorporation of LSP, Lua, and its extensive plugin system, however the plugins make NeoVim brittle and kind of messy. My search for an alternative brought me to Helix, which served as my main editor for six months. However, I found myself missing the rich functionality of NeoVim.

This led me to an unexpected discovery: VS Code, when combined with the [vscode-neovim](https://github.com/vscode-neovim/vscode-neovim) extension, offered a close to ideal blend of NeoVim's command-line efficiency and VS Code's organized framework. This integration effectively mitigates the plugin chaos seen in NeoVim.

Reflecting on these experiences, I propose that Helix should focus on its core strengths of stability and simplicity, possibly by developing a complementary product like 'HelixPlus', as well as having a VS Code plugin similar to NeoVim's. This would allow for advanced, plugin-heavy workflows by runniang Helix in the background, thus preserving its integrity while catering to more complex user needs.

The aim is to ensure Helix evolves by offering a minimalist core and optional extensions for advanced functionality, thus appealing to a broader spectrum of users without compromising its foundational principles.

**Comment by pascalkuthe** [comment, marked as answer]
we are adding a schemne based plugin system, this has been discussed to death.

Note that you seems to fall into the trap as viewing opensoruce as a product. We don't care about popularity or covering every random persons preference. This is not a commercial product that tries to gain market share. We are building an editor that covers our (and likeminded peoples) needs
